common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -361,8 +361,8 @@ public int removeBlocks(String appId, String execId, String[] blockIds) {
     return numRemovedBlocks;
   }
 
-  public Map<String, String[]> getLocalDirs(String appId, String[] execIds) {
-    return Arrays.stream(execIds)
+  public Map<String, String[]> getLocalDirs(String appId, Set<String> execIds) {
```

Comment:
Do we need this function signature change?

Reply:
@mridulm suggested to make this change here:
https://github.com/apache/spark/pull/32140#discussion_r645290580

This change is not necessary for push-based shuffle but it was a small change so I was fine with it.

Reply:
This is not a public api, is it @dongjoon-hyun ?
We could keep the signature and internally use a `Set` if it is being externally depended on.

Reply:
+CC @dongjoon-hyun any thoughts on the above ? Thx